File: lib/languages/python/modules.py
# File: math.py
# Exported functions
def add(a, b):
   return a + b

def subtract(a, b):
   return a - b

# Private function (by convention, not enforced)
def _square(x):
   return x * x

# File: main.py
from math import add
# or import math

def main():
   result = add(5, 3)  # 8
   # math._square(4)  # Works, but not recommended (private by convention)

File: lib/languages/go/modules.go
// File: math/calculator.go
package math

// Add adds two numbers and returns the result
func Add(a, b int) int {
    return a + b
}

// Subtract subtracts b from a and returns the result
func Subtract(a, b int) int {
    return a - b
}

// Private function, not exported
func multiply(a, b int) int {
    return a * b
}

// File: main.go
package main

import (
    "fmt"
    "myapp/math"
)

func main() {
    sum := math.Add(5, 3)
    fmt.Println("Sum:", sum) // Output: Sum: 8
}

